src/views/Requests/CreateRequests.js

```diff
@@ -0,0 +1,299 @@
+/* eslint-disable no-nested-ternary */
+import React, { Component } from 'react';
+import {
+  Container, Row, Form, Button,
+} from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { Spinner } from '@simply007org/react-spinners';
+import _ from 'lodash';
+import BreadCrumbTrail from '../../components/tools/BreadCrumbs';
+import Destination from '../../components/requests/Destination';
+import { sendRequest } from '../../actions/requestActions';
+import { getBookings } from '../../actions/bookingActions';
+import { getLocations } from '../../actions/locationActions';
+import AlertComponent from '../../components/global/AlertComponent';
+
+class CreateRequests extends Component {
+  state = {
+    destinationComponents: [],
+    request: {
+      typeId: null,
+      locationId: null,
+      departureDate: null,
+      returnDate: null,
+      destinations: [],
+    },
+    tripTypes: [
+      {
+        id: 1,
+        name: 'One Way',
+      },
+
+      {
+        id: 2,
+        name: 'Return Trip',
+      },
+      {
+        id: 3,
+        name: 'Multi city',
+      },
+    ],
+    addDestinationComponentDisabled: true,
+    finalMarked: false,
+    oneStopTrip: true,
+  }
+
+  componentDidMount = () => {
+    const { bookings, getBookings, getLocations } = this.props;
+    getBookings();
+    getLocations();
+  }
+
+  makeRequest = () => {
+    const { sendRequest } = this.props;
+
+    const { request } = this.state;
+
+    const requestCopy = { ...request };
+
+    if (requestCopy.returnDate === '') {
+      delete requestCopy.returnDate;
+    }
+    sendRequest(requestCopy);
+  }
+
+  markFinal = async () => {
+    await this.setState({
+      finalMarked: true,
+      addDestinationComponentDisabled: true,
+    });
+  };
+
+  unMarkFinal = async () => {
+    this.setState({
+      finalMarked: false,
+      addDestinationComponentDisabled: false,
+    });
+  };
+
+  addRequestDestination = async (values) => {
+    const { request } = this.state;
+
+    console.log(request);
+    let valuesCopy = { ...values };
+
+    valuesCopy = { ...valuesCopy, bookingId: parseInt(valuesCopy.bookingId, 10) };
+
+    // console.log('===> Our state', this.state);
+
+    if (valuesCopy.departureDate === '') {
+      delete valuesCopy.departureDate;
+    }
+
+    if (request.typeId === '1' || request.typeId === '2') {
+      this.markFinal();
+    }
+
+    if (request.typeId === '1') {
+      const state = {
+        request: _.omit(this.state.request, 'returnDate'),
+      };
+      this.setState(state, () => {
+        // console.log(this.state);
+      });
```

Comment:
Remove this setState because you are not using it

Reply:
This line is no longer in the file. I don't know why it's still showing